import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/ImportAdapter.java

```diff
@@ -110,14 +122,39 @@ public static void main(String[] args) throws IOException, GeneralSecurityExcept
         new DicomWebClient(requestFactory, cstoreDicomwebAddr, cstoreDicomwebStowPath);
     }
 
-    Map<DestinationFilter, IDicomWebClient> destinationMap = configureDestinationMap(
-        flags.destinationConfigInline, flags.destinationConfigPath, credentials);
+    Pair<ImmutableList<Pair<DestinationFilter, IDicomWebClient>>,
+         ImmutableList<Pair<DestinationFilter, AetDictionary.Aet>>> multipleDestinations = configureMultipleDestinationTypesMap(
+        flags.destinationConfigInline, flags.destinationConfigPath, null, credentials);
 
     BackupUploadService backupUploadService = configureBackupUploadService(flags);
 
     DicomRedactor redactor = configureRedactor(flags);
+
+    final IDestinationClientFactory destinationClientFactory;
+    if (flags.sendToAllMatchingDestinations) {
+      destinationClientFactory = new MultipleDestinationClientFactory(
+          multipleDestinations.getLeft(),
+          multipleDestinations.getRight(),
+          defaultCstoreDicomWebClient);
+    } else {
+      destinationClientFactory = new SingleDestinationClientFactory(
+          configureDestinationMap(
+              flags.destinationConfigInline, flags.destinationConfigPath, credentials),
+          defaultCstoreDicomWebClient);
+    }
+    String cstoreSubAet = flags.dimseCmoveAET.equals("") ? flags.dimseAET : flags.dimseCmoveAET;
+    if (cstoreSubAet.isBlank()) {
+      throw new IllegalArgumentException("cstoreSubAet cannot be empty. Please set ");
+    }
+
+    CStoreSenderFactory cStoreSenderFactory = new CStoreSenderFactory(cstoreSubAet);
+    MultipleDestinationUploadService multipleDestinationSendService = new MultipleDestinationUploadService(
+        cStoreSenderFactory,
+        backupUploadService,
+        flags.persistentFileUploadRetryAmount);
+
     CStoreService cStoreService =
-        new CStoreService(defaultCstoreDicomWebClient, destinationMap, redactor, flags.transcodeToSyntax, backupUploadService);
+        new CStoreService(destinationClientFactory, redactor, flags.transcodeToSyntax, multipleDestinationSendService);
```

Comment:
multipleDestinationSendService should only be passed into the constructor if the user is using multiple destinations. Otherwise this should be null, because the multiple destination send service will not work without a backup

Reply:
Maybe multipleDestinationSendService  is a bad name, but this service take a part in single mode processing (send_to_all_matching_destinations not set, but backup present ). In this mode SingleDestinationClientFactory delivers to multipleDestinationSendService  only one healthDestination from destMap after filtering. dicomDest list is empty. --> multipleDestinationSendService  processes healthDestination list with one dest.  (it already works in CStore tests, and now i have fixed it in ImportAdapter.)